### PR TITLE
SRV-2695: upgrade Elixir and use Jason for ExOps

### PR DESCRIPTION
**Pull request by KevinMarth**
[Card](https://rentpath.atlassian.net/browse/SRV-2695)

### CODE REVIEW

File: lib/version.ex [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-  Provides information about curent and previous deployments
+  @moduledoc """
+  Provides information about current and previous deployments
   """
-  defstruct [:deployment, :host]
-  @type t :: %__MODULE__{deployment: map(), host: String.t()}
 
   alias ExOps.{BuildDetails, HostInfo}
 
+  @derive Jason.Encoder
```
Comment (Stratus3D) [inline lib/version.ex:8]: Why do we use `@derive` here, and in `BuildDetails` provide our own implementation? Is there a difference between these two structs?

File: mix.exs [hunk 2]
```diff
@@ -4,8 +4,8 @@ defmodule ExOps.Mixfile do
   def project do
     [
       app: :ex_ops,
-      version: "2.1.0",
-      elixir: "~> 1.8",
+      version: "2.2.0",
+      elixir: "~> 1.10",
```
Comment (Stratus3D) [inline mix.exs:8]: The `.tool-versions` file still specifies elixir `1.9.2`, can you update it as well?

**Review by Stratus3D — CHANGES_REQUESTED** [review]
Please update the Elixir version in the `.tool-versions` file.

File: mix.exs (hunk 2 above)
Comment (KevinMarth) [reply mix.exs:8]: Updated to reflect the Elixir 1.11 upgrade.

File: lib/version.ex (hunk 1 above)
Comment (KevinMarth) [reply lib/version.ex:8]: See https://github.com/rentpath/ex_ops/blob/master/lib/build_details.ex#L8-L14. The `previous` key is added to the parent `BuildDetails` struct after it has been defined to hold a child `BuildDetails`, but the child `BuildDetails` does not have a `previous` key. So, trying to add a `previous` key and `@derive` to the `defstruct` for `BuildDetails` does not work.

The `Version` struct is used simply, so `@derive` works here.

**Review by Stratus3D — APPROVED** [review]

**Review by pasha — APPROVED** [review]
LGTM 👍

**Review by Ebtoulson — APPROVED** [review]

**Pull request merged**